spock-core/src/main/java/org/spockframework/runtime/SpockNode.java

```diff
@@ -126,4 +130,34 @@ protected static ExecutionMode toExecutionMode(org.spockframework.runtime.model.
       .map(er -> new ExclusiveResource(er.getKey(), toLockMode(er.getMode())))
       .collect(Collectors.toSet());
   }
+
+  @Override
+  public Set<TestTag> getTags() {
+    // Lazy initialisation
+    if (tags == null) {
+      tags = new HashSet<>();
+      SpecInfo specInfo;
+      // Node info can be either FeatureInfo or SpecInfo
+      if (getNodeInfo() instanceof FeatureInfo) {
+        FeatureInfo featureInfo = (FeatureInfo) getNodeInfo();
+        addTestTagsFrom(featureInfo.getTags());
+        specInfo = featureInfo.getSpec();
+      }
+      else {
+        specInfo = (SpecInfo) getNodeInfo();
+      }
+      while (specInfo != null) {
```

Comment:
🤔 thinking about it more, it would probably make more sense to delegate the aggregation to the extension completely, and just return the tags that are defined on the node. Meaning the extension will take care of adding the tag to all the necessary nodes.